fairlearn/reductions/_moments/utility_parity.py

```diff
@@ -282,7 +282,8 @@ def signed_weights(self, lambda_vec):
                     (-self.ratio) * event_select / self.prob_event[event] + \
                         group_event_select / self.prob_group_event[event, group]
         if self.M is not None:
-            return self.utility_diff * self.M.dot(lambda_vec)
+            M, lambda_vec = self.M.align(lambda_vec, axis=1, fill_value=0, copy=False)
```

Comment:
Did this come up in a situation when lambda_vec.index was a subset of M.columns or was it some other situation?

Reply:
I think it 's a subset.

self.M.columns is
`[           ('+', 'all', 1),('+', 'all', 2),('+', 'all', 3),('+', 'all', 5),('+', 'all', 6),('+', 'all', 7),('+', 'all', 8),('+', 'all', 9),('-', 'all', 1),('-', 'all', 2),('-', 'all', 3),('-', 'all', 5),('-', 'all', 6),('-', 'all', 7),('-', 'all', 8),('-', 'all', 9)]`

lambda_vec.index
```
MultiIndex([('+', 'all', 1),('+', 'all', 2),('+', 'all', 3),('+', 'all', 6),('+', 'all', 8),('+', 'all', 9),('-', 'all', 1),('-', 'all', 2),('-', 'all', 3),('-', 'all', 6),('-', 'all', 8),('-', 'all', 9)],
           names=['sign', 'event', 'group_id'])
```